### ISSUE TITLE
Design `if-diff` command line tool

### ISSUE THREAD

**Issue by jmcook1186** [Green-Software-Foundation/if#623]
## User story

As a user who needs to compare manifests files I would find it very useful to have a command line tool that can rapidly report whether two manifests are identical and if not how they differ.

## Rationale
We can create a convenience tool for users to quickly see how manifests differ. This tool needs to be designed and specified before we start building. This ticket covers designing and documenting the tool prior to building.
The tool should be able to categorize differences, something along the lines of  'identical', `tree match`, `context-match` `no-overlap` or similar. The user should be able to get granular detail about where and how the manifests differ, similar to `git diff`.

## Implementation details
- [ ] create design proposal for `if-diff`
- [ ] share design with team for approval
- [ ] create documentation and design spec
- [ ] create new ticket for building to the spec

## Priority
3/5

## Size
L

## What does "done" look like?
Design created and signed off, documentation/spec exists

## Deadline
tbc

**Comment by zanete** [comment]
I would propose including the design phase as part of workflow of this issue - https://github.com/Green-Software-Foundation/if/issues/637, closing this to remove noise
